pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -420,6 +420,15 @@ def _get_distributed_sampler(
         """This function is used to created the distributed sampler injected within the user DataLoader."""
         kwargs["shuffle"] = shuffle and not overfit_batches
         kwargs.setdefault("seed", int(os.getenv("PL_GLOBAL_SEED", 0)))
+        if getattr(self.trainer.strategy, "uneven_inputs_support", False) and mode == RunningStage.TRAINING:
```

Comment:
I would much prefer not to do it like this and instead try the approach here https://github.com/PyTorchLightning/pytorch-lightning/issues/11756